src/lorarmesh.cpp

```diff
@@ -14,25 +31,60 @@ LoraMesh::LoraMesh(DataQueue<message_t>* txQueue,
   this->txQueue = txQueue;
   this->rxQueue = rxQueue;
   this->router = router;
+
+  Layer1 = new Layer1Class();
+  Layer1->setPins(LORA_CS, LORA_RST, LORA_IRQ);
+  Layer1->setTxPower(TX_POWER);
+  Layer1->setLoRaFrequency(LORA_FREQ);
+
+  int attempts = MAX_CONNECT_RETRIES;
+
+  while (!Layer1->init()) {
+    delay(CONNECT_DELAY_MILLS);
+    if (attempts-- == 0)
+      ESP.restart();
+  }
+  LL2 = new LL2Class(Layer1);
+  LL2->setInterval(10000);  // set to zero to disable routing packets
+  LL2->init();
+
   xTaskCreatePinnedToCore(task, "LoraMeshTransceiver", 10000, this, 0,
                           &transceiverTaskHandle, LORA_TASKS_CORE);
 }
 
 void LoraMesh::transmit() {
   auto message = txQueue->poll();
+
   if (message == nullptr)
     return;
 
+  LL2->daemon();
+  int datagramsize = 0;
+  struct Datagram datagram;
   if (message->type == CONTROL_MESSAGE) {
-    // TODO broadcast message over LoRaLayer2
+    memcpy(datagram.destination, BROADCAST_NODES, ADDR_LENGTH);
+    datagram.type = message->type;
+    memcpy(datagram.message, &message->data.control, sizeof(control_data_t));
+    datagramsize = DATAGRAM_HEADER;
+    datagramsize += sizeof(control_data_t);
+
   } else if (message->type == DATA_MESSAGE) {
     auto destinationAddr = router->getGatewayAddress();
-    if (destinationAddr == 0)
-      goto freeData;
+    if (destinationAddr > 0) {
+      datagram.type = message->type;
+      memcpy(datagram.destination, &destinationAddr, ADDR_LENGTH);
+      memcpy(datagram.message, message->data.layer2.payload,
+             message->data.layer2.length);
+      datagramsize = DATAGRAM_HEADER;
+      datagramsize += message->data.layer2.length;
 
-    // TODO send to destinationAddr through LoRaLayer2
+    } else {
+      goto freeData;
```

Comment:
Try to not use _goto_.

Reply:
We use a boolean to check if the data should be write ant we remove the goto.